src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -339,6 +342,32 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     }
 
     @Override
+    public List<PartitionEndStatistics> loadTopicEndStatistics(final Collection<Timeline> timelines)
```

Comment:
If I understood correctly, all these timelines must relate to the same repository. So it means that the caller needs to be aware of it.

Thinking about future users of this method, it could be useful to write a comment, warning them of this fact. Even though it should be expected, since this method is in the repository layer, it's bound to a single repo.